### PR TITLE
Azure steps

### PR DESCRIPTION
**Pull request by kenazk**
Azure VM steps for [delete, deallocate, redeploy, restart, list], Disk steps for [list, delete]

### CODE REVIEW

**Comment by kenazk** [comment]
@bradhe can you take a look?

File: azure-disks-list-disks/docs/v1.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+|| `secret` | string | An Azure Service Principal secret. You can pass this into Relay as a secret. See the example below. | None | True |
+|| `tenant_id` | string | An Azure Service Principal tenant_id. You can pass the ID into Relay as a secret. See the example below. | None | True |
+|| `subscription_id` | string | An Azure Subscription ID. You can pass the ID into Relay as a secret. See the example below. | None | True |
+| `resource_group` || string | Resource group to look up disks under | None | False | 
+
+## Outputs
+
+| Name | Data type | Description |
+|------|-----------|-------------|
+| `disks` | array of mappings | The Azure disks in a given subscription. |
```
Comment (bradhe) [inline azure-disks-list-disks/docs/v1.md:22]: Can you get more specific on what the structure of the mapping is? I think it'd be a good habit to get in to and make it easier to use this in the future.

File: azure-disks-list-disks/docs/v1.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+## Specification
+
+| Setting | Child setting | Data type | Description | Default | Required |
+|---------|---------------|-----------|-------------|---------|----------|
+| `azure` || mapping | A mapping of Azure account configuration. | None | True |
+|| `client_id` | string | An Azure Service Principal client id. You can pass the ID into Relay as a secret. See the example below. | None | True |
+|| `secret` | string | An Azure Service Principal secret. You can pass this into Relay as a secret. See the example below. | None | True |
+|| `tenant_id` | string | An Azure Service Principal tenant_id. You can pass the ID into Relay as a secret. See the example below. | None | True |
+|| `subscription_id` | string | An Azure Subscription ID. You can pass the ID into Relay as a secret. See the example below. | None | True |
```
Comment (bradhe) [inline azure-disks-list-disks/docs/v1.md:15]: You'll want to make sure that this aligns with the team working on connections in terms of what they expect to collect from the user.

File: azure-disks-list-disks/step.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+else: 
+  disks = compute_client.disks.list()
+
+# Append disks to list of disks 
+for disk in disks:
+  list_of_disks.append(disk.as_dict())
+
+# Setting output variable `disks` to list of Azure disks
+logging.info('Found {0} disks in the Subscription {1}'.format(len(list_of_disks), subscription_id))
+logging.info('Setting output `disks` to list of {0} found disks'.format(len(list_of_disks)))
```
Comment (bradhe) [inline azure-disks-list-disks/step.py:45]: This log message probably belongs in `debug` verbosity?

File: azure-vms-list-vms/step.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+# Append VMs to list of VMs as dicts
+logging.info('Found the following Azure Virtual Machines:')
+for virtual_machine in virtual_machines:
+  logging.info("{0} {1} {2} {3}".format(virtual_machine.name, virtual_machine.location, virtual_machine.hardware_profile.vm_size, virtual_machine.tags))
+  list_of_vms.append(virtual_machine.as_dict())
+
+# Setting output variable `vms` to list of Azure Virtual Machines
+logging.info('Found {0} Virtual Machines'.format(len(list_of_vms)))
+
+logging.info('Setting output `virtual_machines` to list of {0} found virtual machines'.format(len(list_of_vms)))
```
Comment (bradhe) [inline azure-vms-list-vms/step.py:49]: Similar feedback re: verbosity here.

File: azure-disks-list-disks/docs/v1.md (hunk 2 above)
Comment (kenazk) [reply azure-disks-list-disks/docs/v1.md:15]: @woodburndesigns ^^

**Pull request merged**